kebechet/kebechet_runners.py

```diff
@@ -0,0 +1,171 @@
+    analysis_id: Optional[str] = None,
+):
+    """Run Kebechet by specifying URL of repository to run on."""
+    scheme, _, host, _, slug, _, _ = urllib3.util.parse_url(url)
+    slug = slug[1:]
+    namespace = slug.split("/")[0]
+    project = slug.split("/")[1]
+
+    service_url = f"{scheme}://{host}"
+
+    run(
+        service_type=service,
+        service_url=service_url,
+        namespace=namespace,
+        project=project,
+        parsed_payload=parsed_payload,
+        metadata=kebechet_metadata,
+        analysis_id=analysis_id,
+    )
+
+
+def run_analysis(analysis_id: str, origin: str, service: str) -> None:
+    """Run result managers (meant to be triggered automatically)."""
+    scheme, _, host, _, slug, _, _ = urllib3.util.parse_url(origin)
+    slug = slug[1:]
+    namespace = slug.split("/")[0]
+    project = slug.split("/")[1]
+
+    service_url = f"{scheme}://{host}"
+
+    run(
+        service_type=service,
+        service_url=service_url,
+        namespace=namespace,
+        project=project,
+        enabled_managers=["thoth-provenance", "thoth-advise"],
+        analysis_id=analysis_id,
+    )
+
+
+def run(
+    service_type: str,
+    namespace: str,
+    project: str,
+    service_url: Optional[str],
+    enabled_managers: List[str] = list(REGISTERED_MANAGERS.keys()),
+    parsed_payload: Optional[Dict[Any, Any]] = None,
+    metadata: Optional[Dict[str, Any]] = None,
+    analysis_id: Optional[str] = None,
+) -> None:
+    """Run Kebechet using provided YAML configuration file."""
+    token = os.getenv(f"{service_type}_KEBECHET_TOKEN")
+
+    ogr_service = create_ogr_service(
+        service_type=service_type,
+        service_url=service_url,
+        token=token,
+        github_app_id=os.getenv("GITHUB_APP_ID"),
+        github_private_key_path=os.getenv("GITHUB_PRIVATE_KEY_PATH"),
+    )
+
+    with download_kebechet_config(ogr_service, namespace, project) as f:
+        config = _Config.from_file(f)
+
+    slug = f"{namespace}/{project}"
+
+    managers = config.managers
+
+    for manager in managers:
+        # We do pops on dict, which changes it. Let's create a soft duplicate so if a user uses
+        # YAML references, we do not break.
+        manager = dict(manager)
+        try:
+            manager_name = manager.pop("name")
+        except Exception:
+            _LOGGER.exception(
+                "No manager name provided in configuration entry for %r, ignoring entry",
+                slug,
+            )
+            continue
+
+        if manager_name not in enabled_managers:
+            _LOGGER.debug(
+                "Skipping manager %r because it is not in the list of enabled managers.",
+                manager_name,
+            )
+            continue
+
+        kebechet_manager = REGISTERED_MANAGERS.get(manager_name)
+        if not kebechet_manager:
+            _LOGGER.error("Unable to find requested manager %r, skipping", manager_name)
```

Comment:
For future improvements - we should open in issue to the corresponding repo to have this reported to users.

Reply:
I'll open an issue regarding this.

Reply:
https://github.com/thoth-station/kebechet/issues/726